epsagon/wrappers/python_function.py

```diff
@@ -67,12 +68,41 @@ def wrap_python_function(func, args, kwargs):
             pass
 
 
-def python_wrapper(func):
-    """Epsagon's general python wrapper."""
+def python_wrapper(*args, **kwargs):
+    """
+    Epsagon's general Python wrapper.
+
+    Receives optional keyword arg 'name': used to identify this resource in
+        the application (defaults to function name).
+
+    Options for using:
+    -   @python_wrapper(name='my-resource')
+        def my_function(params):
+            ...
+
+    -   @python_wrapper()
+        def my_function(params):
+            ...
+
+    -   @python_wrapper
+        def my_function(params):
+            ...
+
+    NOTE: Should not be used for two functions in the same stack trace, or when
+          running under a traced context (e.g. AWS Lambda, HTTP frameworks, etc)
+    """
+    name = kwargs.get('name')
+
+    def _inner_wrapper(func):
+
+        @functools.wraps(func)
+        def _python_wrapper(*args, **kwargs):
+            epsagon.trace.trace_factory.get_or_create_trace().prepare()
+            return wrap_python_function(func, args, kwargs, name=name)
```

Comment:
Was this a bug before? Weird...

Reply:
No, this change is needed to support using the decorator both as `@python_wrapper` (for backward compatibility) and `@python_wrapper()`